### PR TITLE
[JENKINS-34008] Add a hook listener for the global library

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-34008](https://issues.jenkins-ci.org/browse/JENKINS-34008)

Yup, it's back but this time against the new repo. =)

Existing rebuild test validates this change.

cc @reviewbybees

### CODE REVIEW

**Comment by svanoort** [comment]
> Yup, it's back but this time against the new repo. =)

The global library can run but it can't hide? :laughing:

**Comment by abayer** [comment]
ping @jglick - like I said on the old PR, what needs to be done for the APIs in question to be considered public?

**Comment by jglick** [comment]
At a minimum, `WorkflowLibRepository.workspace()` could be `public` and have Javadoc explaining what you will find there, pointing you to the listener, etc.

**Comment by abayer** [comment]
Okiedokie.

File: src/main/java/org/jenkinsci/plugins/workflow/cps/global/WorkflowLibRepositoryListener.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/global/WorkflowLibRepositoryListener.java]: Sorry to be a pest but can you use the MIT license please, as specified in the POM?

Comment (abayer) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/global/WorkflowLibRepositoryListener.java]: Yipes, Intellij got silly. Fixing.

**Comment by jglick** [comment]
:bee:

**Pull request merged**